lib/kfuzztest/kfuzztest_examples.c

```diff
@@ -0,0 +1,51 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+#ifndef KFUZZTEST_EXAMPLES_H
+#define KFUZZTEST_EXAMPLES_H
+
+#include <linux/kfuzztest.h>
+
+struct nested_buffers {
+	const char *a;
+	size_t a_len;
+	const char *b;
+	size_t b_len;
+};
+
+/**
+ * The KFuzzTest input format specifies that struct nested buffers should
+ * be expanded as:
+ *
+ * | a | b | pad[8] | *a | pad[8] | *b |
+ *
+ * In DISTINCT mode, this will result in 3 distinct kmalloc'd regions, and
+ * in POISONED mode, the buffer is untouched but the padding will be poisoned.
+ *
+ * In this test case, we look to see that a KASAN warning is triggered in both
+ * cases when overflowing on *a by one.
+ */
+FUZZ_TEST(test_overflow_on_nested_buffer, struct nested_buffers)
+{
+	u32 a_size;
+	u32 b_size;
+
+	KFUZZTEST_EXPECT_NOT_NULL(nested_buffers, a);
+	KFUZZTEST_EXPECT_NOT_NULL(nested_buffers, b);
+	KFUZZTEST_ANNOTATE_LEN(nested_buffers, a_len, a);
+	KFUZZTEST_ANNOTATE_LEN(nested_buffers, b_len, b);
+
+	volatile char c;
```

Comment:
Nit: checkpatch.pl will bark here. You can ignore it (because you are not using a volatile for synchronization) or you can replace `c = arg->a[i]` with a READ_ONCE macro, which should serve the same purpose of preventing the optimizations.
(make sure it triggers a KASAN error)